### PR TITLE
doing homework13

### PR DESCRIPTION
**Pull request by Hromy-Ann**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,6 +12,19 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>${commons-lang3.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.10.2</version>
```
Comment (alexshamrai) [inline pom.xml]: move to properties, stick to one style, please

File: src/test/java/org/example/homework13/blackjack/DeckTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        Deck deck = new Deck();
+        Card card = deck.drawCard();
+        assertEquals(Rank.TWO, card.getRank());
+        assertEquals(Suit.CLUBS, card.getSuit());
+    }
+
+    @Test
+    void shuffle() {
+        Deck referenceDeck = new Deck();
+        Deck shuffledDeck = new Deck();
```
Comment (alexshamrai) [inline src/test/java/org/example/homework13/blackjack/DeckTest.java]: no sense to create 2 Decks. shuffle method changes the state of the field `cards` of the Deck instance. so it is better to check this field before shuffle and after it
So you should check cards list before shuffle and cards list after shuffle

File: src/test/java/org/example/homework13/blackjack/DeckTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        shuffledDeck.shuffle();
+        boolean diffFound = false;
+        for (int i = 0; i < 52; i++) {
```
Comment (alexshamrai) [inline src/test/java/org/example/homework13/blackjack/DeckTest.java]: when you will compare two listst of cards(see previous comment) you won't need to use loop, just compare 2 listst

**Review by alexshamrai — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Hromy-Ann) [reply pom.xml]: fixed

File: src/test/java/org/example/homework13/blackjack/DeckTest.java (hunk 2 above)
Comment (Hromy-Ann) [reply src/test/java/org/example/homework13/blackjack/DeckTest.java]: I had to add a getter for cards list. Updated to use a copy of list as reference

File: src/test/java/org/example/homework13/blackjack/DeckTest.java (hunk 3 above)
Comment (Hromy-Ann) [reply src/test/java/org/example/homework13/blackjack/DeckTest.java]: removed loop

File: src/test/java/org/example/homework13/blackjack/DeckTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        assertEquals(Rank.TWO, card.getRank());
+        assertEquals(Suit.CLUBS, card.getSuit());
+    }
+
+    @Test
+    void shuffle() {
+        Deck deck = new Deck();
+        List<Card> referenceList = new ArrayList<>(deck.getCards());
+        deck.shuffle();
+        assertNotEquals(referenceList, deck.getCards(), "Deck was not shuffled");
```
Comment (alexshamrai) [inline src/test/java/org/example/homework13/blackjack/DeckTest.java]: only assert equals is not enough. ideally do something like this
```suggestion
        var deck = new Deck();
        List<Card> originalCards = new ArrayList<>(deck.getCards());
        deck.shuffle();
        List<Card> shuffledCards = deck.getCards();
        assertEquals(originalCards.size(), shuffledCards.size(),
                "The size of the deck should remain unchanged");
        assertNotEquals(originalCards, shuffledCards,
                "The deck should be shuffled and not in original order");
        assertTrue(shuffledOrder.containsAll(originalCards) && originalCards.containsAll(shuffledCards),
                "The shuffled deck should contain all original cards");```

**Review by alexshamrai — CHANGES_REQUESTED** [review]

**Review by alexshamrai — APPROVED** [review]

**Pull request merged**